packages/echo-core/src/services/baseClient/baseClient.ts

```diff
@@ -85,8 +85,7 @@ export default class BaseClient {
         try {
             const headers = {
                 Authorization: 'Bearer ' + token,
-                ...headerOptions,
-                ...(body && { 'Content-Type': 'application/json' })
+                ...headerOptions
```

Comment:
removing application/json completely is a breaking change, it's better to overwrite it. Something like this, now headerOptions Content-Type will overwrite application/json if it exist in headerOptions
        const headers = {
            Authorization: 'Bearer ' + token,
            ...{ 'Content-Type': 'application/json' }
            ...headerOptions,
        };

Reply:
This method did not work. :/ 
  What about:   ```

> const headers = {
>                 Authorization: 'Bearer ' + token,
>                 ...({ 'Content-Type': 'application/json' } && headerOptions)
>             };
>